### PR TITLE
added gen question script

### PR DESCRIPTION
**Pull request by ilan-pinto**
## Type of change

- [ ] Refactor
- [V] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [V] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: scripts/questions_gen.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+    dir_list = dirs_all_files(question_folder)
+    
+    if len(dir_list) == 0: 
+        raise "couldn't find dirs in questions folder"
+    
+    print("*** starting question iteration ")
+    
+    full_results = []
+    total_correctness_score = []
+    for dir in dir_list:
```
[the remark this answers is not in the document]
Comment (ilan-pinto) [reply scripts/questions_gen.py:169]: @asamal4  I am not sure it's needed. If Qe wants to generate questions only from 1 dir, then he must specify only one directory.

File: scripts/questions_gen.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    Returns:
+        Tuple[float, str]: A tuple containing the score as a float and the reasoning as a string.
+    """
+    
+    print("eval_response:", eval_response)
+    
+    eval_response_parsed = eval_response.split("\n")
+    eval_len =len (eval_response_parsed)
+    
+    if eval_len == 0 :
```
Comment (tisnik) [inline scripts/questions_gen.py:59]: `match` might be better ;)

File: scripts/questions_gen.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        results["correctness"] = res_table
+                
+        end_time = time.time()
+        execution_time_seconds = end_time - start_time
+        print(f"*** Completed correctness evaluation: execution time in min: {execution_time_seconds/60}")
+        full_results.append(results)
+    
+
+
+async def main():
```
Comment (tisnik) [inline scripts/questions_gen.py:218]: why we run it asynchronously? it's one process with one coroutine, right

**Review by tisnik — COMMENTED** [review]
it looks ok

File: scripts/questions_gen.py (hunk 3 above)
Comment (ilan-pinto) [reply scripts/questions_gen.py:218]: llama_index evel runs async.

**Comment by tisnik** [comment]
@ilan-pinto you need to `pdm update the_new_library_you_added_to_pyproject`

Also please run `make format` and then `make verify`

**Comment by tisnik** [comment]
please run `make format`

**Pull request merged**